Mathlib/RingTheory/Artinian.lean

```diff
@@ -513,13 +519,41 @@ lemma primeSpectrum_finite : {I : Ideal R | I.IsPrime}.Finite := by
     inf_le_right.eq_of_not_lt (H (p ⊓ s.inf Subtype.val) ⟨insert p s, by simp⟩)
   rwa [← Subtype.ext <| (@isMaximal_of_isPrime _ _ _ _ q.2).eq_of_le p.2.1 hq2]
 
-variable (R) in
+variable (R)
 /--
 [Stacks Lemma 00J7](https://stacks.math.columbia.edu/tag/00J7)
 -/
 lemma maximal_ideals_finite : {I : Ideal R | I.IsMaximal}.Finite := by
   simp_rw [← isPrime_iff_isMaximal]
   apply primeSpectrum_finite R
 
+local instance : Finite {I : Ideal R | I.IsMaximal} := (maximal_ideals_finite R).to_subtype
+
+noncomputable local instance (I : {I : Ideal R | I.IsMaximal}) : Field (R ⧸ I.1) :=
+  have := mem_setOf.mp I.2; Ideal.Quotient.field I.1
+
+/-- The quotient of a commutative artinian ring by its nilradical is isomorphic to
+a product of finitely many fields, namely the quotients by the maximal ideals. -/
+noncomputable def quotNilradicalEquivPi :
+    R ⧸ nilradical R ≃+* ∀ I : {I : Ideal R | I.IsMaximal}, R ⧸ I.1 :=
+  .trans (Ideal.quotEquivOfEq <| ext fun x ↦ by simp_rw [mem_nilradical,
+    nilpotent_iff_mem_prime, Submodule.mem_iInf, Subtype.forall, isPrime_iff_isMaximal, mem_setOf])
+  (Ideal.quotientInfRingEquivPiQuotient _ fun I J h ↦
+    Ideal.isCoprime_iff_sup_eq.mpr <| I.2.coprime_of_ne J.2 <| by rwa [Ne, Subtype.coe_inj])
+
+/-- Any reduced commutative artinian ring is isomorphic to a product of
```

Comment:
add this in the initial docstring?

Reply:
There are other results worth mentioning in the module docstring, e.g. all prime ideals are maximal and there are only finitely many, the Jacobson radical is nilpotent ... Maybe use a dedicated documentation PR for all of them?

Reply:
Well, I'd suggest to just do it right now for the stuff you add here, and leave to somebody else to do the rest (might be you if you wish!).

Reply:
added docstring to Artinian and SimpleModule